aiida/tools/dbimporters/plugins/matp.py

```diff
@@ -0,0 +1,217 @@
+        self._mpr = MPRester(self._api_key)
+
+    def setup_db(self, **kwargs):
+        """
+        Setup the required parameters to the REST API
+
+        :param api_key: the API key to be used to access MAPI
+        """
+        api_key = kwargs['api_key']
+        if api_key is None:
+            try:
+                self._api_key = os.environ['PMG_MAPI_KEY']
+            except KeyError:
+                raise ValueError('API key not supplied and PMG_MAPI_KEY environment '
+                                 'variable not set. Either pass it when initializing the class, '
+                                 'or set the environment variable PMG_MAPI_KEY to your API key.')
+        self._api_key = api_key
+        self._verify_api_key()
+
+    def _verify_api_key(self):
+        """
+        Verify the supplied API key by issuing a request to Materials Project.
+        """
+        response = requests.get(
+            'https://www.materialsproject.org/rest/v1/api_check', headers={'X-API-KEY': self._api_key})
+        response_content = response.json()  # a dict
+        if 'error' in response_content:
+            raise RuntimeError(response_content['error'])
+        if not response_content['valid_response']:
+            raise RuntimeError('Materials Project did not give a valid response for the API key check.')
+        if not response_content['api_key_valid']:
+            raise RuntimeError('Your API key for Materials Project is not valid.')
+
+    @property
+    def api_key(self):
+        """
+        Return the API key configured for the importer
+        """
+        return self._api_key
+
+    @property
+    def collection(self):
+        """
+        Return the collection that will be queried
+        """
+        return self._collection
+
+    @property
+    def get_supported_keywords(self):
+        """
+        Returns the list of all supported query keywords
+
+        :return: list of strings
+        """
+        return self._supported_keywords
+
+    def query(self, **kwargs):
+        """
+        Query the database with a given dictionary of query parameters for a given collection
+
+        :param query: a dictionary with the query parameters
+        :param collection: the collection to query
+        """
+        try:
+            query = kwargs['query']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `query` as a key. This '
+                                 'should contain a dictionary with the right query needed.')
+        try:
+            collection = kwargs['collection']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `collection` as a key.')
+
+        if not isinstance(query, dict):
+            raise TypeError('The query argument should be a dictionary')
+
+        if collection is None:
+            collection = self._collection
+
+        if collection == 'structure':
```

Comment:
``self._collection`` is defined as ``structures`` and here the check is for ``structure``. Is this correct?

Reply:
Fixed.